src/scenes/gameuiscreen.ts

```diff
@@ -1,28 +1,51 @@
 import { ScreenHeight, ScreenWidth } from "core/main";
 import Phaser from "phaser";
 import { log } from "utilities/logger";
-import { interactify } from "utilities/utils";
+import { assert, interactify } from "utilities/utils";
+import { float, int, Nullable } from "../definitions/utils";
+import { Tile } from "../entities/Tile";
+import TileManager from "../managers/TileManager";
 
 export class GameUiScreen extends Phaser.Scene {
+    private back: Nullable<Phaser.GameObjects.Rectangle>;
+
     public constructor() {
         super("GameUiScreen");
+        this.back = null;
```

Comment:
this.back is the complete grey rectangle area? Would be great to have a more elaborate variable name like this.itemBox or this.uiBackground

Reply:
+1